Library/Formula/h2o.rb

```diff
@@ -0,0 +1,82 @@
+class H2o < Formula
+  homepage "https://github.com/h2o/h2o/"
+  url "https://github.com/h2o/h2o/archive/v0.9.0.tar.gz"
+  sha1 "37b84750900cdb56c3be574477d6f3327d92a4d7"
+  head "https://github.com/h2o/h2o.git"
+
+  option "with-libuv", "Build the H2O library as well as the executable."
+
+  depends_on "cmake" => :build
+  depends_on "libyaml"
+  depends_on "openssl"
+  depends_on "libuv" => :optional
+
+  def install
+    system "cmake", ".", *std_cmake_args
+
+    if build.with? "libuv"
+      system "make", "libh2o"
+      lib.install "libh2o.a"
+    end
+
+    system "make", "install"
+
+    mkdir_p etc/"h2o"
+    mkdir_p var/"h2o"
+    (etc+"h2o").install "examples/h2o/alternate.crt", "examples/h2o/alternate.key"
+    (etc+"h2o").install "examples/h2o/server.crt", "examples/h2o/server.key"
+    (var+"h2o").install "examples/doc_root/index.html"
+    (etc+"h2o/h2o.conf").write conf_example
+  end
+
+  def conf_example; <<-EOS.undent
+    listen: 8080
+    listen:
+       port: 8081
+       ssl:
+         certificate-file: #{etc}/h2o/server.crt
+         key-file: #{etc}/h2o/server.key
+    hosts:
+      "127.0.0.1.xip.io:8080":
+        paths:
+          /:
+            file.dir: #{var}/h2o/
+    EOS
+  end
```

Comment:
Yeah, it's a modified example. Essentially, if I used the example I'd have had to inreplace several lines to fit our folder formatting, and rather than messing around with inreplacing an example it seemed easier & more flexible to just write a short one ourselves. The example is written to be run inside the tarball, which isn't overly useful to us.

The SSL support is in the example, I just modified the paths on that. Happy to remove the SSL element though and reduce the amount we have to manually ship into etc here, and minimise the config file a little more.

Sent from my iPhone

> On 13 Jan 2015, at 11:12, Mike McQuaid <notifications@github.com> wrote:
> 
> In Library/Formula/h2o.rb:
> 
> > +  end
> > +
> > +  def conf_example; <<-EOS.undent
> > +    listen: 8080
> > +    listen:
> > +       port: 8081
> > +       ssl:
> > +         certificate-file: #{etc}/h2o/server.crt
> > +         key-file: #{etc}/h2o/server.key
> > +    hosts:
> > +      "127.0.0.1.xip.io:8080":
> > +        paths:
> > +          /:
> > +            file.dir: #{var}/h2o/
> > +    EOS
> > +  end
> Where is this from? Is it a modified example? I reckon to scrap the caveats we just ship the example configuration (if there's one provided) without SSL support.
> 
> —
> Reply to this email directly or view it on GitHub.
>